ansible_ai_connect/ai/api/utils/segment.py

```diff
@@ -181,6 +181,10 @@ def redact_seated_users_data(event: Dict[str, Any], allow_list: Dict[str, Any])
     Returns:
     - dict: A new dictionary containing only the allowed nested keys from the source dictionary.
     """
+    # If an asterisk is found in the allow list, just return the given event dict to allow all keys.
+    if "*" in allow_list:
```

Comment:
It'd be nice to have a test for these with your new check?

I _know_ SonarCloud says you have 100%... but is that a false positive? I don't _see_ any.

Reply:
I have added two lines to test_operational_telemetry()`
```
        self.user.rh_user_has_seat = True
        self.user.organization = Organization.objects.get_or_create(id=1)[0]
```
which forces to run the test case with a "seated" user.  Without this allow_list change, the test case failed. Isn't this  change enough for the test coverage?

Reply:
> Isn't this change enough for the test coverage?

I found there were some test cases on redact operations in `test_segment.py`.  I'll add two test cases to it.

Reply:
Yes, that's fine @TamiTakamiya 

It wasn't clear from your PR.

Reply:
Hey @TamiTakamiya, @manstis, sorry for late response, just my 2c the idea behind allow list agains block list in this case is making sure you are logging only something you really need to log, and avoid situation we are storing things we are not supposed to store.

Whit the asterisk it is easier to make a mistake and store unintended data.

Reply:
@hasys You're never late to the party and, as author of the original white listing, your opinion is important.. When you wrote the enhancement you no doubt had a clear vision in mind and it should not be _bastardized_ to fit other use-cases (as this is how we end up with spaghetti code!)

What would you have proposed?

IIUC the issue was that the Chat Bot events were not recognised as "valid" events and hence filtered out.

@TamiTakamiya's change means they are no longer blocked by the filter and obviates the need to list all fields.